CoreScriptsRoot/Modules/ChatSelector.lua

```diff
@@ -0,0 +1,162 @@
+--[[
+	// FileName: ChatSelector.lua
+	// Written by: Xsitsu
+	// Description: Code for determining which chat version to use in game.
+]]
+local FORCE_CorescriptNewLoadChat 	= false		-- Force FFLag on client to be true always
+local FORCE_GetShouldUseLuaChat 	= false 	-- Force SFFlag value read from server to be true always
+
+
+local CoreGuiService = game:GetService("CoreGui")
+local RobloxGui = CoreGuiService:WaitForChild("RobloxGui")
+
+local StarterGui = game:GetService("StarterGui")
+local UserInputService = game:GetService("UserInputService")
+local GuiService = game:GetService("GuiService")
+
+local Util = require(RobloxGui.Modules.ChatUtil)
+
+local function GetUseLuaFlag()
+	local loop_continue = true
+	while loop_continue do
+		local success, retVal = pcall(function()
+			return game.IsSFFlagsLoaded
+		end)
+		if not success then
+			loop_continue = false
+		elseif retVal then
+			loop_continue = false
+		else
+			wait(0.1)
+		end
+	end
+
+	local success, retVal = pcall(function() return game:GetService("Chat"):GetShouldUseLuaChat() end)
+	local useNewChat = success and retVal
+	return useNewChat
+end
+
+local readFlagSuccess, flagEnabled = pcall(function() return settings():GetFFlag("CorescriptNewLoadChat") end)
+local TryLoadNewChat = readFlagSuccess and flagEnabled
+
+
+local useModule = nil
+
+local state = {Visible = true}
+local interface = {}
+do
+	function interface:ToggleVisibility()
+		if (useModule) then
+			useModule:ToggleVisibility()
+		else
+			state.Visible = not state.Visible
+		end
+	end
+
+	function interface:SetVisible(visible)
+		if (useModule) then
+			useModule:SetVisible(visible)
+		else
+			state.Visible = visible
+		end
+	end
+
+	function interface:FocusChatBar()
+		if (useModule) then
+			useModule:FocusChatBar()
+		else
```

Comment:
I wanted to keep it in there like that to keep the same format that the other functions had and to blatantly say to not do anything in this situation.